### PR TITLE
Cleanup MinIOInstance Controller

### PR DESCRIPTION
**Pull request by nitisht**
This PR adds support for a new controller to handle MirrorInstance CRD. We already had a controller to handle MinIOInstance CRD. With two controllers, changed the file layout to make it idiomatic to Operator concept.

- Update k8s.io/client-go version to v0.18.2
- Add MirrorInstance Controller

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,10 +54,16 @@ kubectl create -f https://raw.githubusercontent.com/minio/minio-operator/master/
...
-kubectl patch minioinstances.miniocontroller.min.io minio --patch "$(cat examples/patch.yaml)" --type=merge
+kubectl patch minioinstances.miniooperator.min.io minio --patch "$(cat examples/patch.yaml)" --type=merge
 ```
 
-You can further keep adding new zones in the `patch.yaml` file and apply the patch, to add new nodes to existing cluster. 
+You can further keep adding new zones in the `patch.yaml` file and apply the patch, to add new nodes to existing cluster.
+
+**NOTE**: Important point to consider _before_ using cluster expansion:
+
+Ensure every Persistent Volume bound to MinIO Operator PVC's has reclaim policy as `retain`. If you attempt cluster expansion while the Persistent Volume reclaim policy is set to something else, it may lead to data loss. If you have the reclaim policy set to something else, change it as explained in [Kubernetes documents](https://kubernetes.io/docs/tasks/administer-cluster/change-pv-reclaim-policy/).
```
Comment (harshavardhana) [inline README.md:64]: Can this be elaborated a bit better from the point of view that PV creation is an important step?

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ MinIO is a high performance distributed object storage server, designed for larg
 
 ### Prerequisites
 
-- Kubernetes version v1.15.5 and above.
+- Kubernetes version v1.17.0 and above for compatibility. MinIO Operator uses `k8s/client-go` v0.17.0.
```
Comment (krisis) [inline README.md:15]: we are using v0.18.0 - https://github.com/minio/minio-operator/pull/85/files#diff-37aff102a57d3d7b797f152915a6dc16R15

File: minio-operator.yaml [hunk 3]
```diff
@@ -132,7 +147,7 @@ spec:
       serviceAccountName: minio-operator-sa
       containers:
         - name: minio-operator
-          image: minio/k8s-operator:1.0.8
+          image: nitishtiwari/operator:mirror
```
Comment (krisis) [inline minio-operator.yaml:150]: is this left behind for testing during development?

File: README.md (hunk 1 above)
Comment (nitisht) [reply README.md]: Reworded the para, PTAL

File: minio-operator.yaml (hunk 3 above)
Comment (nitisht) [reply minio-operator.yaml]: yes, fixed.

**Comment by nitisht** [comment]
> This also looks like a breaking change @nitisht ?

This PR specifically is not breaking, although the previous PR #80 added the `zones` field which may cause issue for existing users and they may have to update their yaml files.

**Comment by nitisht** [comment]
Comments are addressed, PTAL @harshavardhana @krisis

**Pull request merged**